### PR TITLE
Schedule nupcol

### PR DESCRIPTION
**Pull request by joakim-hove**
Internalize the nupcol property in the new `ScheduleState`class.

Downstream: https://github.com/OPM/opm-simulators/pull/3004

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3004 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3004 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3004 please

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3004 please

File: src/opm/parser/eclipse/EclipseState/Schedule/ScheduleState.cpp [hunk 1]
```diff
@@ -65,17 +65,26 @@ const PAvg& ScheduleState::pavg() const {
...
+
+int ScheduleState::nupcol() const {
+    return this->m_nupcol;
+}
 
 bool ScheduleState::operator==(const ScheduleState& other) const {
     return this->m_start_time == other.m_start_time &&
            this->m_tuning == other.m_tuning &&
            this->m_end_time   == other.m_end_time;
+           this->m_nupcol == other.m_nupcol;
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/ScheduleState.cpp:80]: This statement has no effect.  You need to add an `&&` operator on the previous line too.

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/ScheduleState.cpp:80]: Rebase gone bad - fixed

**Comment by joakim-hove** [comment]
jenkins build this opm-simulators=3004 please

**Pull request merged**